src/transform/transform.ts

```diff
@@ -293,34 +293,60 @@ function recursiveAddMarshallingFormat(schema: Schema, marshallingFormat: 'json'
 }
 
 // we will transform operation request parameter schema types to Go types
-function processOperationRequests(session: Session<CodeModel>) {
+async function processOperationRequests(session: Session<CodeModel>) {
   // pre-process multi-request operations as it can add operations to the operations
   // collection, and iterating over a modified collection yeilds incorrect results
   for (const group of values(session.model.operationGroups)) {
     for (const op of values(group.operations).toArray()) {
       if (op.language.go!.description) {
         op.language.go!.description = parseComments(op.language.go!.description);
       }
-      if (op.requests!.length > 1) {
+
+      const normalizeOperationName = await session.getValue('normalize-operation-name', false);
+      
+      if (!normalizeOperationName){
+        // previous operation naming logic: keep original name if only one body type, and add suffix for operation with non-binary body type if more than one body type
+        if (op.requests!.length > 1) {
+          for (const req of values(op.requests)) {
+            const newOp = <Operation>{...op};
+            newOp.language = clone(op.language)
+            newOp.requests = (<Array<Request>>op.requests).filter(r => r === req);
+            let name = op.language.go!.name;
+            // for the non-binary media types we create a new method with the
+            // media type name as a suffix, e.g. FooAPIWithJSON()
+            if (req.protocol.http!.knownMediaType !== KnownMediaType.Binary) {
+              let suffix: string;
+              switch (req.protocol.http!.knownMediaType) {
+                case KnownMediaType.Json:
+                  suffix = 'JSON';
+                  break;
+                case KnownMediaType.Xml:
+                  suffix = 'XML';
+                  break;
+                default:
+                  suffix = capitalize(req.protocol.http!.knownMediaType);
+              }
+              name = name + 'With' + suffix;
+            }
+            newOp.language.go!.name = name;
+            newOp.language.go!.protocolNaming = new protocolMethods(newOp.language.go!.name);
+            group.addOperation(newOp);
+            if (req.language.go!.description) {
+              req.language.go!.description = parseComments(req.language.go!.description);
+            }
+          }
+          group.operations.splice(group.operations.indexOf(op), 1);
+        }
+      } else {
```

Comment:
A lot of this code looks duplicated from the above branch.  Can this be consolidated?  The algorithm itself looks good.

Reply:
Resolved.